lib/share_progress/button_client.rb

```diff
@@ -0,0 +1,40 @@
+require 'share_progress'
+
+module ShareProgress
+  class ButtonClient
+    def initialize
+      @connection = ShareProgress::ConnectionFactory.get_connection
+    end
+
+    def new
+    end
+
+    def update
+    end
+
+    def find(id)
+      @connection.make_call endpoint('read'), 'get', { id: id }
+    end
+
+    def destroy(id)
+      @connection.make_call endpoint('delete'), 'post', { id: id }
+    end
+
+    def all(options={})
+      acceptable = [:offset, :limit]
+      @connection.make_call endpoint, 'get', filter_keys(options, acceptable)
+    end
```

Comment:
I know it's early days, so hope you don't mind if I suggest a tweak to your API. I think your methods' intentions would be clearer if you used `httparty` directly:

```ruby
# lib/share_progress/client.rb
module ShareProgress
  class Client
    include HTTParty

    base_uri 'run.shareprogress.org/api/v1'
    default_params api_key: ENV['API_KEY']
  end
end

# lib//share_progress/button.rb
module ShareProgress
  class Button

   # It's a GET request, so use query in place of body
   # Named arguments ensure the passed options meet your
   # acceptable check, but also can optionally provide default values:
   # button.all
   # or
   # button.all( offset: 100, limit: 300)
   #
    def all(offset: 5, limit: 100)
      Client.get '/buttons', query: { offset: offset, limit: limit }
    end

  end
end
```
I don't see a benefit to accessing `httparty` via `@connection.make_call`, but perhaps I'm missing something.

I'm just sharing some ideas. Let me know what you think.